python/cuml/cuml/ensemble/randomforest_common.pyx

```diff
@@ -63,150 +111,177 @@ def compute_max_features(
-        """Translate sklearn-style criterion string to cuML equivalent"""
-        if criterion is None:
-            split_criterion = cls._default_split_criterion
-        else:
-            if criterion == "squared_error":
-                split_criterion = "mse"
-            elif criterion == "absolute_error":
-                split_criterion = "mae"
-            elif criterion == "poisson":
-                split_criterion = "poisson"
-            elif criterion == "gini":
-                split_criterion = "gini"
-            elif criterion == "entropy":
-                split_criterion = "entropy"
-            else:
-                raise NotImplementedError(
-                    f'Split criterion {criterion} is not yet supported in'
-                    ' cuML. See'
-                    ' https://docs.rapids.ai/api/cuml/nightly/api.html#random-forest'
-                    ' for full information on supported criteria.'
-                )
-        return criterion
-
-    @device_interop_preparation
-    def __init__(self, *, split_criterion, n_streams=4, n_estimators=100,
-                 max_depth=16, handle=None, max_features='sqrt', n_bins=128,
-                 bootstrap=True,
-                 verbose=False, min_samples_leaf=1, min_samples_split=2,
-                 max_samples=1.0, max_leaves=-1, accuracy_metric=None,
-                 dtype=None, output_type=None, min_weight_fraction_leaf=None,
-                 n_jobs=None, max_leaf_nodes=None, min_impurity_decrease=0.0,
-                 min_impurity_split=None, oob_score=None, random_state=None,
-                 warm_start=None, class_weight=None,
-                 criterion=None,
-                 max_batch_size=4096, **kwargs):
-
-        sklearn_params = {"min_weight_fraction_leaf": min_weight_fraction_leaf,
-                          "max_leaf_nodes": max_leaf_nodes,
-                          "min_impurity_split": min_impurity_split,
-                          "oob_score": oob_score, "n_jobs": n_jobs,
-                          "warm_start": warm_start,
-                          "class_weight": class_weight}
-
-        for key, vals in sklearn_params.items():
-            if vals and not cuml.accel.enabled():
-                raise TypeError(
-                    " The Scikit-learn variable ", key,
-                    " is not supported in cuML,"
-                    " please read the cuML documentation at "
-                    "(https://docs.rapids.ai/api/cuml/nightly/"
-                    "api.html#random-forest) for more information")
-
-        for key in kwargs.keys():
-            if key not in self._param_names and not cuml.accel.enabled():
-                raise TypeError(
-                    " The variable ", key,
-                    " is not supported in cuML,"
-                    " please read the cuML documentation at "
-                    "(https://docs.rapids.ai/api/cuml/nightly/"
-                    "api.html#random-forest) for more information")
```

Comment:
Before we had a superset of parameters `cuml` actually supported in `__init__` - however, we'd error on init if any that we didn't support were present. We now only include parameters we support, and let the python interpreter raise the `TypeError` for parameters that aren't supported. Much simpler.